app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt

```diff
@@ -1339,15 +1331,8 @@ class MainViewModel @Inject constructor(
      */
     private fun getActionFolderRole(threads: List<Thread>?, message: Message? = null): FolderRole? {
         return when {
-            message != null -> {
-                message.folder.role
-            }
-            threads?.firstOrNull()?.folderId == FolderController.SEARCH_FOLDER_ID -> {
-                folderController.getFolder(threads.first().folderId)?.role
-            }
-            else -> {
-                threads?.firstOrNull()?.folder?.role
-            }
+            message != null -> message.folder.role
+            else -> threads?.firstOrNull()?.folderRole
         }
```

Comment:
It looks like it won't get the correct folder role when the thread is from the Search.